tests/disaster-recovery/mdr/test_app_failover_and_relocate.py

```diff
@@ -0,0 +1,226 @@
+
+        # Application Failover to Secondary managed cluster
+        secondary_cluster_name = get_current_secondary_cluster_name(
+            workload.workload_namespace
+        )
+        if config.RUN.get("mdr_failover_via_ui"):
+            logger.info("Start the process of Failover from ACM UI")
+            config.switch_acm_ctx()
+            failover_relocate_ui(
+                acm_obj,
+                workload_to_move=f"{workload.workload_name}-1",
+                policy_name=workload.dr_policy_name,
+                failover_or_preferred_cluster=secondary_cluster_name,
+            )
+        else:
+            failover(
+                failover_cluster=secondary_cluster_name,
+                namespace=workload.workload_namespace,
+            )
+
+        # Verify application are running in other managedcluster
+        # And not in previous cluster
+        set_current_primary_cluster_context(workload.workload_namespace)
+        wait_for_all_resources_creation(
+            workload.workload_pvc_count,
+            workload.workload_pod_count,
+            workload.workload_namespace,
+        )
+
+        # Verify the failover status from UI
+        if config.RUN.get("mdr_relocate_via_ui"):
+            config.switch_acm_ctx()
+            verify_failover_relocate_status_ui(acm_obj)
+
+        # Start nodes if cluster is down
+        wait_time = 60
+        if primary_cluster_down:
+            logger.info(
+                f"Waiting for {wait_time} minutes before starting nodes of previous primary cluster"
+            )
+            time.sleep(60)
+            nodes_multicluster[primary_cluster_index].start_nodes(node_objs)
+            logger.info(
+                "Waiting for some minutes after starting nodes of previous primary cluster"
+            )
+            time.sleep(120)
+            wait_for_nodes_status([node.name for node in node_objs])
+            logger.info(
+                "Wait for all the pods in openshift-storage to be in running state"
+            )
+            assert wait_for_pods_to_be_running(
+                timeout=720
+            ), "Not all the pods reached running state"
+
+        # Verify application are deleted from old cluster
+        set_current_secondary_cluster_context(workload.workload_namespace)
+        wait_for_all_resources_deletion(workload.workload_namespace)
+
+        # Validate data integrity
+        set_current_primary_cluster_context(workload.workload_namespace)
+        validate_data_integrity(workload.workload_namespace)
+
+        # Unfenced the managed cluster which was Fenced earlier
+        enable_unfence(drcluster_name=self.drcluster_name)
+
+        # Reboot the nodes which unfenced
+        gracefully_reboot_ocp_nodes(workload.workload_namespace, self.drcluster_name)
+
+        # Application Relocate to Primary managed cluster
+        secondary_cluster_name = get_current_secondary_cluster_name(
```

Comment:
Run IOs for some time before relocate

Reply:
Is it done?

Reply:
As discussed in call, we don't have VR concept in MDR so additional wait for sync is not required. So marking them resolved

Reply:
ack